### ISSUE TITLE
How to change default cst cell into our own data ? 

### ISSUE THREAD

**Issue by P705** [e0404/matRad#383]
Hello

I'm a freshman of radiotherapy field. So I don't understand how to change the code related to the cst cell information.

Actually, i want create my own cell and use in matrad.
How can i do this work ?

Thanks

**Comment by wahln** [comment]
Hi,

the cst holds information about organ and target segmentation, i.e., their names, type (OAR/TARGET), list of voxel indices in the ct, and the set of optimization objectives / constraints. Some additional information for display and biology is also included (see also our wiki page https://github.com/e0404/matRad/wiki/The-cst-cell).

Do you want to create your own organ / target segmentation from scratch or what is your goal?

**Comment by P705** [comment]
I have series of dicom file and wanna load them in the matRad.  
How can i create organ and target segmentation ?
create my own cst struct and analysis the diom data.

**Comment by wahln** [comment]
Do you have your structures as DICOM (RTStruct)? Or do you only have your CT? Our DICOM importer (matRad_importDicomGUI.m) should be able to import Segmentations from RTStruct.
If you want to delineate/create them from scratch, matRad currently does not feature a graphical way to delineate your own volumes. However, https://github.com/e0404/matRad/blob/master/examples/matRad_example1_phantom.m gives an example to set up ct and cst with Target and OAR from scratch based on a simple phantom.

**Comment by P705** [comment]
Can you explain about RTStruct and how can i create that ?

**Comment by wahln** [comment]
RTStruct is a DICOM file specification for storing segmentations. If you don't have that with your DICOM data, you would need other software that allows you to delineate structures and export them to RTStruct (maybe MITK or SlicerRT can do this). Or, alternatively, binary masks (i.e. image cubes in ct resolution that have ones where the volume is supposed to be, and zeros otherwise) which can be imported manually in Matlab or in nrrd format with our binary file import functions (in folder IO).

**Comment by P705** [comment]
Hi
I have a series of DICOM with RTStruct, but can't impotrt them in the matRad.
The window open for importing dicom, don't have any key for do that.
I attach the dicom folder and guide me to import them in matRad.
[dicom.zip](https://github.com/e0404/matRad/files/3897411/dicom.zip)

**Comment by wahln** [comment]
Before I continue to help: 
Is your uploaded DICOM-data appropriately anonymized? If not, it needs to be deleted immediately!! We can not work with and share non-anonymized patient data. 

For now I can only give a generic response that could help you: you need to click the Patient ID at least once to show up all respective SOPInstanceUIDs in the selected folder.

**Comment by P705** [comment]
I delete the uploaded data as you say.
But, I don't understand your explanation about click on patient data and all respective SOPInatanxeUIDs.
Please, more explain about this subject.

**Comment by mingersming** [comment]
It takes 6 clicks to import your own DICOM files. 

![Screenshot - 28 11 2019 , 10_32_35](https://user-images.githubusercontent.com/34025665/69795230-10bc5500-11cc-11ea-81b3-6ea48ce07bd5.png)

At step 2 you have to specify the folder where your DICOM files are located.
At step 6 you save the DICOM files as a *.mat file. 

This file can now be loaded with matRad ("Load *.mat data") for further purpose.

**Comment by markbangert** [comment]
Thank you @mingersming for this brilliant explanation :)
